src/bin/pg_autoctl/keeper.c

```diff
@@ -1965,3 +1966,93 @@ keeper_config_accept_new(Keeper *keeper, KeeperConfig *newConfig)
 	return config_accept_new_ssloptions(&(config->pgSetup),
 										&(newConfig->pgSetup));
 }
+
+
+/*
+ * reload_configuration reads the supposedly new configuration file and
+ * integrates accepted new values into the current setup.
+ */
+bool
+keeper_reload_configuration(Keeper *keeper, bool firstLoop)
```

Comment:
I also would like to be able to retry the SIGHUP processing, but it's pretty hard to invent a solid approach that would not surprise anybody. As you're saying we can't really retry the SIGHUP action infinitely. I think the current implementation is easy to understand and makes sense.

Also we have many other function that return a bool and that are used in context where we can't really act on their return value. I prefer to leave it that way, so that if we come up with a better error handling strategy later, we have the information we need already.